lib/dfe/analytics/event.rb

```diff
@@ -73,7 +73,8 @@ def with_entity_table_name(table_name)
       end
 
       def with_data(hash)
-        @event_hash.deep_merge!(data: hash_to_kv_pairs(hash))
+        @event_hash.deep_merge!(data: hash_to_kv_pairs(hash[:data]))
+        @event_hash.deep_merge!(hidden_data: hash_to_kv_pairs(hash[:hidden_data]))
```

Comment:
I think we might need to  only add `data` or `hidden_data` if it is present, as the initialise and checksum events don't add `hidden_data`
```suggestion
        @event_hash.deep_merge!(data: hash_to_kv_pairs(hash[:data])) if hash[:data].any?
        @event_hash.deep_merge!(hidden_data: hash_to_kv_pairs(hash[:hidden_data]))  if hash[:hidden_data].any?
```